server/src/main/java/org/opensearch/index/codec/fuzzy/MurmurHash64.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.fuzzy;
+
+import org.apache.lucene.util.BitUtil;
+import org.apache.lucene.util.BytesRef;
+
+/**
+ * Utility to calculate hash for incoming bytes.
+ */
+public class MurmurHash64 {
```

Comment:
Please note that this is an implementation of the older MurmurHash2 which is inferior to MurmurHash3 in terms of collision resistance ([more info](https://news.ycombinator.com/item?id=4036841)).

I suggest using the already available [MurmurHash3](https://github.com/opensearch-project/OpenSearch/blob/c0852f84f0647f78a45e71396d6370a39ba4922f/server/src/main/java/org/opensearch/common/hash/MurmurHash3.java#L44) hash function, or better, use the [T1ha](https://github.com/opensearch-project/OpenSearch/blob/3a8bbe9095ec028e74ec9a2700bf417ca5ed2fb5/libs/common/src/main/java/org/opensearch/common/hash/T1ha1.java#L39) hash function I had introduced recently which offers much better performance. Here's [more info](https://github.com/opensearch-project/OpenSearch/pull/8788#issuecomment-1676398213) on it.

Reply:
Thanks for the pointers on the existing hash functions which have been recently added in OpenSearch. 

I had tried with MurMurHash3 and did not see a visible impact during macro-benchmarks with the 2 hash functions. Given it still makes sense to reuse existing code instead of adding new implementation, I'm checking if we can use `T1ha` and replace accordingly.